riptide-core/src/test/java/org/zalando/riptide/DynamicBaseUrlTest.java

```diff
@@ -0,0 +1,72 @@
+package org.zalando.riptide;
+
+import java.net.URI;
+import java.util.function.Supplier;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.web.client.MockRestServiceServer;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.HttpStatus.Series.SUCCESSFUL;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
+import static org.zalando.riptide.Bindings.on;
+import static org.zalando.riptide.Navigators.series;
+import static org.zalando.riptide.Route.pass;
+
+public class DynamicBaseUrlTest {
+
+    private final Rest unit;
+    private final MockRestServiceServer server;
+
+    @SuppressWarnings("unchecked")
+    private final Supplier<URI> baseUrlProviderMock = mock(Supplier.class);
```

Comment:
Is that really important for your test?`If yes, I'd suggest to use a second test for that.